changeme/scanners/telnet.py

```diff
@@ -1,29 +1,78 @@
 from .scanner import Scanner
 import telnetlib
-
+import re
+import time
 
 class Telnet(Scanner):
 
-    def __init__(self, cred, target, config, username, password):
+    def __init__(self, cred, target, username, password, config):
         super(Telnet, self).__init__(cred, target, config, username, password)
 
     def _check(self):
         try:
-            telnet = telnetlib.Telnet(str(self.target), int(self.port), int(self.config.timeout))
-            telnet.read_until("login: ")
+            telnet = telnetlib.Telnet(str(self.target.host))
+            timeout_allowed = int(self.cred['auth']['blockingio_timeout'])
+            wait_for_pass_prompt = int(self.cred['auth']['telnet_read_timeout'])
+
+            retval = telnet.open(str(self.target.host), int(self.target.port), timeout=timeout_allowed)
+            retval._has_poll = False    # telnetlib hackery :)
```

Comment:
What's this line do? I keep getting an error:

    [21:43:45][telnet][_check] Error: 'NoneType' object has no attribute '_has_poll'